### PR TITLE
GERO-241 - handle pandas.NA input as null and add null tests for GSC Report integration edge cases.

### PR DESCRIPTION
**Pull request by dustinbleile**
Add some tests for the edgecase null types found in integration testing.

### CODE REVIEW

**Comment by creisle** [comment]
Rather than replacing the NA here it might make more sense to just do a single df.fillna('') before you create the JSON?

File: tests/test_inputs.py [hunk 1]
```diff
@@ -18,6 +21,7 @@
 from ipr.util import logger
 
 DATA_DIR = os.path.join(os.path.dirname(__file__), 'test_data')
+NULLS = ['', None, np.nan, pd.NA]
```
Comment (creisle) [inline tests/test_inputs.py:24]: It can be dangerous to assume '' is always a null value. We often use null to mean not given/specified or unknown but then sometimes '' is the actual value. For example the untemplated sequence between breakpoints in a structural variant or the ref/alt seq in a small mutation

File: tests/test_inputs.py [hunk 2]
```diff
@@ -59,27 +63,59 @@ def test_maintains_optional_fields(self):
...
+                small_mut[key] = null
+                records = preprocess_small_mutations([small_mut])
+                record = records[0]
+                assert record['variantType'] == 'mut'
+                for col in original:
+                    assert col in record
+                assert record['variant'] == 'A1BG:p.V460M'
+                assert 'endPosition' in record
+
+    def test_load_small_mutations_probe(self) -> None:
```
Comment (creisle) [inline tests/test_inputs.py:87]: I think i must be missing something here. Why is this test indented and given the self argument? it does not appear to be part of a class

**Review by creisle — CHANGES_REQUESTED** [review]

File: tests/test_inputs.py (hunk 2 above)
Comment (dustinbleile) [reply tests/test_inputs.py:87]: Part of 'class TestPreProcessSmallMutations:'
Code view just looks weird

File: tests/test_inputs.py (hunk 1 above)
Comment (dustinbleile) [reply tests/test_inputs.py:24]: For the keys being tested here, I think all the NULLS are equivalent.
Maybe just a comment about some of the exceptions like the ref/alt seq?
Would you list of  exceptions?

Comment (creisle) [reply tests/test_inputs.py:24]: ya, I would definitely put a comment about not using this for fields where empty string is valid, I might rename the variable to match but i can't think of anything other than `NON_EMPTY_STRING_NULLS` and that's a bit long so maybe just the comment for now unless you have a better name

File: tests/test_inputs.py [hunk 3]
```diff
@@ -59,27 +63,59 @@ def test_maintains_optional_fields(self):
...
+        records = preprocess_copy_variants(
+            pd.read_csv(os.path.join(DATA_DIR, 'copy_variants.tab'), sep='\t').to_dict('records')
+        )
+        assert records
+        assert len(records) == 4603
+        assert records[0]['variantType'] == 'cnv'
+        assert 'variant' in records[0]
+
+    def test_null(self):
+        for kb_cat in list(INPUT_COPY_CATEGORIES.values()) + NULLS:
```
Comment (creisle) [inline tests/test_inputs.py:110]: not blocking but might be nice to use parameterize here instead so that you can easily tell which case fails. If you stack the decorators it will do all combinations

```python
@pytest.mark.parameterize('kb_category', INPUT_COPY_CATEGORIES.values())
@pytest.mark.parameterize('column_name', COPY_OPTIONAL)
@pytest.mark.parameterize('null_value', NULLS)
def test_nulls(kb_category, column_name, null_value): 
    copy_var = {'gene': 'ERBB2', 'kbCategory': kb_category}
    copy_var[column_name] = null
    records = preprocess_copy_variants([copy_var])
    record = records[0]
    assert record['variantType'] == 'cnv'
```

**Review by creisle — APPROVED** [review]
